src/get-started/create-service.md

```diff
@@ -541,116 +569,67 @@ pub struct WalletQuery {
 impl CryptocurrencyApi {
     /// Endpoint for getting a single wallet.
     pub fn get_wallet(
-        state: &ServiceApiState,
-        query: WalletQuery
+        self,
+        state: &ServiceApiState<'_>,
+        pub_key: PublicKey,
     ) -> api::Result<Wallet> {
-        let snapshot = state.snapshot();
-        let schema = CurrencySchema::new(snapshot);
+        let schema = CurrencySchema::new(state.service_data());
         schema
-            .wallet(&query.pub_key)
-            .ok_or_else(|| api::Error::NotFound("\"Wallet not found\"".to_owned()))
+            .wallets
+            .get(&pub_key)
+            .ok_or_else(|| api::Error::not_found().title("Wallet not found"))
     }
 
     /// Endpoint for dumping all wallets from the storage.
     pub fn get_wallets(
-        state: &ServiceApiState,
-        _query: ()
+        self,
+        state: &ServiceApiState<'_>
     ) -> api::Result<Vec<Wallet>> {
-        let snapshot = state.snapshot();
-        let schema = CurrencySchema::new(snapshot);
-        let idx = schema.wallets();
-        let wallets = idx.values().collect();
-        Ok(wallets)
+        let schema = CurrencySchema::new(state.service_data());
+        Ok(schema.wallets.values().collect())
     }
 }
 ```
 
 As with the transaction endpoint, the methods have an idiomatic signature
 `fn(&ServiceApiState, MyQuery) -> api::Result<MyResponse>`.
 
-### Wire API
-
-As the final step of the API implementation, we need to tie the request
-processing logic to the specific endpoints.
-We do this in the `CryptocurrencyApi::wire()` method:
+Also we will declare a helper method to wire API, which later can be invoked
+by the service:
 
 ```rust
 impl CryptocurrencyApi {
-    pub fn wire(builder: &mut ServiceApiBuilder) {
-        // Binds handlers to the specific routes.
+    /// 'ServiceApiBuilder' facilitates conversion between read requests and REST
+    /// endpoints.
+    pub fn wire(self, builder: &mut ServiceApiBuilder) {
```

Comment:
Nit / off-topic: Does consuming `self` have any purpose? Future compatibility?

Reply:
I copy-pasted it from the actual `cryptocurrency` example. IMHO it's better to keep them as similar as possible. Later we can improve both the service and the docs at the same time.

Reply:
That's why I've used "off-topic" marking 🙃 If we're going to change this, naturally the changes would involve the example code as well.